### PR TITLE
ROX-20259: Pin SCCs to Deployments to prevent preemtping SCC

### PR DESCRIPTION
**Pull request by SimonBaeumer**
## Description

This PR pins the SCCs to the Pods to prevent preempting the SCCs by OpenShift.
Additionally the SCCs must be applied before the Deployment to cluster, otherwise the Pod Admission Controller will reject the Pod because the SCC does not exist yet.

Relevant OpenShift documentation: https://docs.openshift.com/container-platform/4.14/authentication/managing-security-context-constraints.html#security-context-constraints-requiring_configuring-internal-oauth

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

 - Install StackRox into an OpenShift cluster, check SCC annotation is present on all Pods in stackrox namespace
 - Added unit tests for Helm charts

### CODE REVIEW

**Comment by vikin91** [comment]
This is a great idea to add the pinning, but let's not pin our custom SCCs that are about to be deleted. Let's use the stock ones instead.
If you think, we should keep delivering custom SCCs, then please share your thoughts on that, as we plan to abandon them.

File: image/templates/helm/stackrox-secured-cluster/templates/sensor.yaml.htpl [hunk 1]
```diff
@@ -27,6 +27,7 @@ spec:
       annotations:
         traffic.sidecar.istio.io/excludeInboundPorts: "8443,9443"
         {{- include "srox.podAnnotations" (list . "deployment" "sensor") | nindent 8 }}
+        openshift.io/required-scc: stackrox-sensor
```
Comment (vikin91) [inline image/templates/helm/stackrox-secured-cluster/templates/sensor.yaml.htpl:30]: Let me pick this example to highlight one thing. The `stackrox-sensor` SCC will be created only if we select a checkbox for operator installs (set a value on helm installs). In all other cases, sensor will run with the stock `restricted-v2` SCC.
Sensor does not need the custom SCC and there is no reason to use it. We should stick to `restricted-v2` wherever possible (this applies also to other services).

Comment (vikin91) [inline image/templates/helm/stackrox-secured-cluster/templates/sensor.yaml.htpl:30]:
```suggestion
        openshift.io/required-scc: restricted-v2
```

File: image/templates/helm/shared/templates/02-scanner-06-deployment.yaml.htpl [hunk 2]
```diff
@@ -12,6 +12,7 @@ metadata:
     {{- include "srox.labels" (list . "deployment" "scanner") | nindent 4 }}
   annotations:
     {{- include "srox.annotations" (list . "deployment" "scanner") | nindent 4 }}
+    openshift.io/required-scc: {{ include "srox.globalResourceName" (list . "stackrox-scanner") }}
```
Comment (vikin91) [inline image/templates/helm/shared/templates/02-scanner-06-deployment.yaml.htpl:15]: I think this SCC is never created - we need to double-check which SCC is used by scanner in reality.

File: image/templates/helm/stackrox-secured-cluster/templates/admission-controller.yaml [hunk 3]
```diff
@@ -26,6 +26,7 @@ spec:
       annotations:
         traffic.sidecar.istio.io/excludeInboundPorts: "8443"
         {{- include "srox.podAnnotations" (list . "deployment" "admission-control") | nindent 8 }}
+        openshift.io/required-scc: stackrox-admission-control
```
Comment (vikin91) [inline image/templates/helm/stackrox-secured-cluster/templates/admission-controller.yaml:29]:
```suggestion
        openshift.io/required-scc: restricted-v2
```

File: image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml [hunk 4]
```diff
@@ -24,6 +24,7 @@ spec:
       labels:
         app: central
         {{- include "srox.podLabels" (list . "deployment" "central") | nindent 8 }}
+        openshift.io/required-scc: {{ include "srox.globalResourceName" (list . "stackrox-central") }}
```
Comment (vikin91) [inline image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml:27]: This still needs to be checked what is the optima value. You can refer to the table in this WIP-PR (when it is updated): https://github.com/stackrox/stackrox/pull/7671

**Comment by SimonBaeumer** [comment]
> This is a great idea to add the pinning, but let's not pin our custom SCCs that are about to be deleted. Let's use the stock ones instead. If you think, we should keep delivering custom SCCs, then please share your thoughts on that, as we plan to abandon them.

Perfect, in that case we do not need to pin our custom SCCs.
The pinning was implemented  for OpenShift 4.15 and must be added to avoid preempting the SCC we expect, in this case `restricted-v2`.
I have written the developer who implemented the SCC pinning if it is a blocker for 4.15.

**Pull request closed without merging**